os/apt.py: Add support for passing --allow-unauthenticated

lamby opened the pull request:
This is useful for packages that bootstrap their own apt-key setup - only
the initial installation will require overriding. Notable examples are the
Dropbox and Google Chrome packages.

(Setting force=yes is far too strong: I only want to bypass
authentication!)

Signed-off-by: Chris Lamb chris@chris-lamb.co.uk

gregdek commented:
Thanks @lamby for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

lamby commented:
Rebased (against `devel` branch) by request.

ready_for_review

gregdek commented:
Thanks @lamby for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

gregdek commented:
Thanks @lamby for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

lamby commented:
> This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed.

They were extremely quick the last time I sent patches :)

(Also, nothing more demotivating to outside contributions to have such long delays btw..)

gregdek commented:
Thanks @lamby for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

lamby commented:
ready_for_review

robynbergeron commented:
Thanks @lamby for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

lamby commented:
> This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Not to be churlish but I sent a bunch of PRs about a year ago and they were literally accepted within hours - is there a new policy or am I confusing core/non-core..? :)

gregdek commented:
Thanks @lamby for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

gregdek commented:
Thanks @lamby for this PR. Unfortunately, it is not mergeable in its current state due to merge conflicts. Please rebase your PR. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

lamby commented:
@gregdek 

ready_for_review

(Again)

resmo commented:
Feature request: https://github.com/ansible/ansible/issues/11620

resmo commented on packaging/os/apt.py:
```diff
@@ -73,6 +73,12 @@
+      - Ignore if packages can't be authenticated. This is useful for packages that bootstrap their own apt-key setup.
+    required: false
+    default: "no"
+    choices: [ "yes", "no" ]
```
since this arg is new in 2.1, add a `version_added: "2.1"` below this line

resmo commented on packaging/os/apt.py:
```diff
@@ -73,6 +73,12 @@
     choices: [ "yes", "no" ]
+  allow_unauthenticated:
+    description:
+      - Ignore if packages can't be authenticated. This is useful for packages that bootstrap their own apt-key setup.
```
IMHO the description is not quite right. If the package is signed, and the key is registered using apt-key, `--allow-unauthenticated` is not needed.

Would you mind change the description to just:

> Ignore if packages can't be authenticated.

mika replied:
@resmo what the "This is useful for packages that bootstrap their own apt-key setup" is supposed to mean from _my_ PoV (me not being the submitter) is that you need `--allow-unauthenticated` for _bootstrapping_ the apt-key setup, like in `apt-get install --allow-unauthenticated foobar-archive-keyring` where `foobar-archive-keyring` provides the apt-key(s).

resmo replied:
I understand. agreed

lamby replied:
Updated.

resmo commented:
I see the need of this feature.

`force=yes` is not the right way for allow-unauthenticated. And I don't see any other easy way to implement it without breaking backwards compatibility. Since we also have an argument for   `only_upgrade` I am +1 for it, when the description has been corrected.

lamby committed:
os/apt.py: Add support for passing --allow-unauthenticated

This is useful for packages that bootstrap their own apt-key setup - only
the initial installation will require overriding. Notable examples are the
Dropbox and Google Chrome packages.

(Setting force=yes is far too strong: I only want to bypass
authentication!)

Signed-off-by: Chris Lamb <chris@chris-lamb.co.uk>

FlorinAndrei commented:
Is this going to make it into version 2.0.2?
